apps/emails-and-messages/src/modules/ui/code-editor.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useCallback, useRef } from "react";
+
+import Editor from "@monaco-editor/react";
+import { useTheme } from "@saleor/macaw-ui";
+
+type Props = {
+  onChange(value: string): void;
+  initialTemplate: string;
+  value: string;
+  language: string;
+};
+
+export const CodeEditor = ({ initialTemplate, onChange, value, language }: Props) => {
+  const { themeType } = useTheme();
+  const editorRef = useRef(null);
+
+  // @ts-ignore
+  function handleEditorDidMount(editor, monaco) {
+    editorRef.current = editor;
+  }
+
+  const handleOnChange = useCallback(
+    (value?: string) => {
+      onChange(value ?? "");
+    },
+    [value]
+  );
+
+  return (
+    <>
+      <Editor
+        height="40vh"
```

Comment:
have you checked how it looks when you make smaller screen? please try 1440x900. Id avoid `vh`